packages/providers/src/util.ts

```diff
@@ -98,23 +98,87 @@ export const calculateTransactionFee = ({
 
 const DEFAULT_BLOCK_EXPLORER_URL = 'https://fuellabs.github.io/block-explorer-v2';
 
+const getPathFromInput = (key: string, value: string | number | undefined) => {
+  switch (key) {
+    case 'address':
+      return `address/${value}`;
+    case 'txId':
+      return `transaction/${value}`;
+    case 'blockNumber':
+      return `block/${value}`;
+    default:
+      return `/${key}/${value}`;
+  }
+};
+
 /**
  * Builds a block explorer url based on and the given path, block explorer URL and provider URL
  */
 export const buildBlockExplorerUrl = ({
   blockExplorerUrl,
   path,
   providerUrl,
+  address,
+  txId,
+  blockNumber,
 }: {
   blockExplorerUrl?: string;
-  path: string;
+  path?: string;
   providerUrl?: string;
+  address?: string;
+  txId?: string;
+  blockNumber?: number;
 }) => {
   const explorerUrl = blockExplorerUrl || DEFAULT_BLOCK_EXPLORER_URL;
 
+  // make sure that only ONE or none of the following is defined: address, txId, blockNumber
+  const customInputParams = [
+    {
+      key: 'address',
+      value: address,
+    },
+    {
+      key: 'txId',
+      value: txId,
+    },
+    {
+      key: 'blockNumber',
+      value: blockNumber,
+    },
+  ];
+  const definedValues = customInputParams
+    .filter((param) => param.value)
+    .map((param) => ({
+      key: param.key,
+      value: param.value,
+    }));
+  if (definedValues.length > 1) {
+    throw new Error(
+      `Only one of the following can be passed in to buildBlockExplorerUrl: ${customInputParams
+        .map((param) => param.key)
+        .join(', ')}`
+    );
+  }
+  if (definedValues.length === 0 && !path) {
+    throw new Error(
+      `One of the following must be passed in to buildBlockExplorerUrl: ${customInputParams
+        .map((param) => param.key)
+        .join(', ')}, path`
+    );
+  }
+  if (path && definedValues.length > 0) {
+    throw new Error(
+      `You cannot pass in a path to buildBlockExplorerUrl along with any of the following: ${customInputParams
+        .map((param) => param.key)
+        .join(', ')}`
+    );
+  }
```

Comment:
Adding fn blocks inside backtick strings can often become hard to read/maintain. Perhaps setting these values on a variable before throwing the error might be cleaner:

```suggestion
    const inputKeys = customInputParams.map(({ key }) => key).join(', ')
    throw new Error(
      `You cannot pass in a path to buildBlockExplorerUrl along with any of the following: ${inputKeys}`
    );
  }
```